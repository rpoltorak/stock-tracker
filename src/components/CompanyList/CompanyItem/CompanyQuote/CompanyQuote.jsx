import React from "react";
import PropTypes from "prop-types";
import { ChangeIndicator } from "../../../ChangeIndicator";

CompanyQuote.propTypes = {
  currency: PropTypes.string.isRequired,
  quote: PropTypes.shape({
    price: PropTypes.string,
    change: PropTypes.string,
    changePercent: PropTypes.string,
    latestTradingDay: PropTypes.latestTradingDay,
  }).isRequired,
};

export function CompanyQuote({ quote, currency }) {
  return (
    <div className="small">
      <span className="font-weight-bold">{quote.price} </span>
      {currency}
      <ChangeIndicator
        change={quote.change}
        changePercent={quote.changePercent}
      />
      Closed: {quote.latestTradingDay}
      {closed}
    </div>
  );
}
