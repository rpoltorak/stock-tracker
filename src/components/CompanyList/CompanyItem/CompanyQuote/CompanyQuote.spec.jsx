import React from "react";
import { render } from "@testing-library/react";
import { CompanyQuote } from "./CompanyQuote";

describe("CompanyQuote", () => {
  test("matches the snapshot", () => {
    const props = {
      currency: "USD",
      quote: {
        price: "312.93",
        change: "0.6200",
        changePercent: "0.2768%",
        latestTradingDay: "2019-10-01",
      },
    };
    const component = render(<CompanyQuote {...props} />);

    expect(component).toMatchSnapshot();
  });
});
