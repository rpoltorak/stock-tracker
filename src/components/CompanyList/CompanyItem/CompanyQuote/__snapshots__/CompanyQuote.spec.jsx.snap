// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyQuote matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="small"
      >
        <span
          class="font-weight-bold"
        >
          312.93
           
        </span>
        USD
        <span
          class="mx-2"
          style="color: green;"
        >
          0.6200
          (
          0.2768%
          )
        </span>
        Closed: 
        2019-10-01
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="small"
    >
      <span
        class="font-weight-bold"
      >
        312.93
         
      </span>
      USD
      <span
        class="mx-2"
        style="color: green;"
      >
        0.6200
        (
        0.2768%
        )
      </span>
      Closed: 
      2019-10-01
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
